tests/import_proxy_test.go

```diff
@@ -0,0 +1,326 @@
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/containerized-data-importer/pkg/common"
+	"kubevirt.io/containerized-data-importer/pkg/controller"
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1beta1"
+)
+
+const (
+	username         = "foo"
+	password         = "bar"
+	httpPort         = "8080"
+	httpPortWithAuth = "8081"
+	proxyServerName  = "cdi-test-proxy"
+)
+
+var _ = Describe("Import Proxy tests", func() {
+
+	f := framework.NewFramework("import-proxy-func-test")
+	tinyCoreIsoHTTP := func() string { return fmt.Sprintf(utils.TinyCoreIsoURL, f.CdiInstallNs) }
+	tinyCoreIsoHTTPS := func() string { return fmt.Sprintf(utils.HTTPSTinyCoreIsoURL, f.CdiInstallNs) }
+	createHTTPDataVolume := func(dataVolumeName, size, url string) *cdiv1.DataVolume {
+		dataVolume := utils.NewDataVolumeWithHTTPImport(dataVolumeName, size, url)
+		return dataVolume
+	}
+	createHTTPDataVolumeWithAuth := func(dataVolumeName, size, url string) *cdiv1.DataVolume {
+		dataVolume := utils.NewDataVolumeWithHTTPImport(dataVolumeName, size, url)
+		stringData := map[string]string{
+			common.KeyAccess: username,
+			common.KeySecret: password,
+		}
+		s, _ := utils.CreateSecretFromDefinition(f.K8sClient, utils.NewSecretDefinition(nil, stringData, nil, f.Namespace.Name, "mysecret"))
+		dataVolume.Spec.Source.HTTP.SecretRef = s.Name
+		return dataVolume
+	}
+	createHTTPSDataVolume := func(dataVolumeName, size, url string) *cdiv1.DataVolume {
+		dataVolume := utils.NewDataVolumeWithHTTPImport(dataVolumeName, size, url)
+		cm, err := utils.CopyFileHostCertConfigMap(f.K8sClient, f.Namespace.Name, f.CdiInstallNs)
+		Expect(err).To(BeNil())
+		dataVolume.Spec.Source.HTTP.CertConfigMap = cm
+		return dataVolume
+	}
+	createHTTPSDataVolumeWithAuth := func(dataVolumeName, size, url string) *cdiv1.DataVolume {
+		dataVolume := utils.NewDataVolumeWithHTTPImport(dataVolumeName, size, url)
+		cm, err := utils.CopyFileHostCertConfigMap(f.K8sClient, f.Namespace.Name, f.CdiInstallNs)
+		Expect(err).To(BeNil())
+		dataVolume.Spec.Source.HTTP.CertConfigMap = cm
+		stringData := map[string]string{
+			common.KeyAccess: username,
+			common.KeySecret: password,
+		}
+		s, _ := utils.CreateSecretFromDefinition(f.K8sClient, utils.NewSecretDefinition(nil, stringData, nil, f.Namespace.Name, "mysecret"))
+		dataVolume.Spec.Source.HTTP.SecretRef = s.Name
+		return dataVolume
+	}
+	createProxyURL := func(isHTTPS bool, withBasicAuth bool) string {
+		var auth string
+		port := httpPort
+		if withBasicAuth {
+			port = httpPortWithAuth
+			auth = fmt.Sprintf("%s:%s@", username, password)
+		}
+		return fmt.Sprintf("http://%s%s.%s:%s", auth, proxyServerName, f.CdiInstallNs, port)
+	}
+
+	FDescribe("importing from cdi proxied web server", func() {
```

Comment:
If you FDescribe (or FDescribeTable or FIt, or FContext) it will focus just those tests and the overall test run will fail. For testing this is what you want, but in the CI we want to run all the tests, so make sure to remove the F(ocus) from the tests you added before pushing.

Reply:
Included it by mistake...